File: src/main/java/us/illyohs/sparkyforge/webhook/SparkyServer.java
package us.illyohs.sparkyforge.webhook;

import us.illyohs.sparkyforge.webhook.github.GitHookManager;

import com.google.common.io.ByteStreams;
import com.google.gson.Gson;
import spark.Request;
import spark.Response;
import spark.Route;
import spark.Spark;

public class SparkyServer
{
//    private int     port    = 3567;
    private static String  hookURL = "/sparkyhook/";

    public Gson gson;

    public SparkyServer(int port)
    {

        Spark.port(port);
        Spark.init();
        Github();
    }


    public static void Github()
    {
        Spark.post(hookURL + "/github", (request, response) -> {
            String sig       = request.headers("X-Hub-Signature");
            String eventType = request.headers("X-GitHub-Event");
            byte[] json      = ByteStreams.toByteArray(request.raw().getInputStream());

            return new GitHookManager(sig, eventType, json);
        });
    }

//    public static handle
}

File: src/main/scala/us/illyohs/sparkyforge/webhook/SparkyServer.scala
package us.illyohs.sparkyforge.webhook

import us.illyohs.sparkyforge.webhook.github.GitHooker

import com.google.common.io.ByteStreams
import spark.{Request, Response, Route, Spark}

object SparkyServer {

  def Github() {
    Spark.post("/github", new Route {
      override def handle(request: Request, response: Response) = {
        val sig = request.headers("X-Hub-Signature")
        val eventType = request.headers("X-GitHub-GitEvent")
        val data:Array[Byte] = ByteStreams.toByteArray(request.raw().getInputStream)

        new GitHooker(sig, eventType, data)

      }
    })
  }
}

class SparkyServer(val port: Int) {
  Spark.port(port)
  Spark.init()
  SparkyServer.Github()
  //    public static handle
}